src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/locations/LocationsReconciliationService.kt

```diff
@@ -248,45 +254,40 @@ class LocationsReconciliationService(
     return null
   }
 
-  fun historyDoesNotMatch(nomis: LocationResponse, dps: Location): Boolean {
-    if (expectedDpsHistorySize(nomis.amendments) == (dps.changeHistory?.size ?: 0)) {
-      return false
+  fun historyDoesNotMatch(nomis: LocationResponse, dps: LegacyLocation): Boolean {
+    val filteredAmendmentResponses = filterAmendmentResponses(nomis.amendments)
+    val filteredDpsHistoryResponses = dps.changeHistory?.filter { it.attribute != "Converted Cell Type" }
+    if ((filteredAmendmentResponses?.size ?: 0) != (filteredDpsHistoryResponses?.size ?: 0)) {
+      return true
     }
-    val nomisListConverted = nomis.amendments!!.map { a ->
+    val nomisListConverted = filteredAmendmentResponses?.map { a ->
       ChangeHistory(toHistoryAttribute(a.columnName), a.amendedBy, a.amendDateTime, a.oldValue, a.newValue)
-    }
+    } ?: emptyList()
     val nomisSetSorted = nomisListConverted.sortedBy { it.amendedDate + it.attribute + it.amendedBy + it.oldValue + it.newValue }
-    val dpsSetSorted = dps.changeHistory!!.sortedBy { it.amendedDate + it.attribute + it.amendedBy + it.oldValue + it.newValue }
+    val dpsSetSorted = filteredDpsHistoryResponses?.sortedBy { it.amendedDate + it.attribute + it.amendedBy + it.oldValue + it.newValue } ?: emptyList()
 
-    var dpsIndex = 0
-    nomisSetSorted.forEach { nomisItem ->
-      val dpsItem = dpsSetSorted.elementAtOrNull(dpsIndex)
-      if (dpsItem != null) {
-        if (nomisItem.attribute == dpsItem.attribute &&
-          nomisItem.amendedBy == dpsItem.amendedBy &&
-          nomisItem.amendedDate == dpsItem.amendedDate &&
-          nomisItem.oldValue == dpsItem.oldValue &&
-          nomisItem.newValue == dpsItem.newValue
-        ) {
-          // matches
-        } else {
-          return true
-        }
-      } else {
-        return true
-      }
-      dpsIndex++
+    if (dpsSetSorted.size > nomisSetSorted.size) {
+      return true
     }
-    return false
+
+    return nomisSetSorted.zip(dpsSetSorted)
+      .any { (nomisItem, dpsItem) ->
+        nomisItem.attribute != dpsItem.attribute ||
+          nomisItem.amendedBy != dpsItem.amendedBy ||
+          nomisItem.amendedDate != dpsItem.amendedDate ||
+          nomisItem.oldValue != dpsItem.oldValue ||
+          nomisItem.newValue != dpsItem.newValue
+      }
   }
 
   private fun reportHistory(nomisList: List<AmendmentResponse>?, dpsList: List<ChangeHistory>?): String {
-    val nomisListConverted = nomisList!!.map { nomis ->
+    val nomisListConverted = filterAmendmentResponses(nomisList)?.map { nomis ->
       ChangeHistory(toHistoryAttribute(nomis.columnName), nomis.amendedBy, nomis.amendDateTime, nomis.oldValue, nomis.newValue)
-    }
+    } ?: emptyList()
 
-    val onlyInDps = dpsList!! - nomisListConverted
-    val onlyInNomis = nomisListConverted - dpsList
+    val dpsListNonNull = dpsList ?: emptyList()
```

Comment:
Nullability is part of the Type of this object - not sure you need to includethe type in the name?